### DISCUSSION TITLE
Allow additions to the Commit Message

### DISCUSSION THREAD

**Discussion by KristofKlein** [microsoft/AL-Go#1140] [Ideas]
Hi,

New to the Al:Go Party :) Still playing around with it and learning a lot ( which is not as easy as I hoped for ). Anyway, a small idea: We use an external tool for Issue tracking (jira). Luckily there is integration for that in github. the only thing it needs: an issue tag in the commit message. It would be nice, If I could add that to the Al:Go workflows as an optional input that gets added to the commit message in the end as a footer.

**Comment by freddydk** [comment]
Hi Kristof 

Could you elaborate a bit more on this?
What workflows are you thinking about?
and is the issue tag a URL or just some arbitrary string?

**Reply by KristofKlein** [reply]
I think it applies for all that have a "direct commit?" option like Update AL-Go System Files.

If it matters I think it would be right now just a string. There happens some simple commit msg parsing for things like XXX-1234 - extensions and integrations of Jira will translate them automatically into links actually.

**Reply by aholstrup1** [reply]
If I understand your problem correctly it sounds very similar to an issue we also have in Microsoft :) We don't use the JIRA integration but instead the Azure Boards integration which can link commits/pull requests/issues by writing "AB#<WorkItemId>" in the description. 

Our idea would be to create an AL-Go setting like this: 
```
"commitOptions": {
    "messageSuffix": "AB#123456",
    "autoMerge": true,
    "pullRequestLabels": ["Automation"]
  }
```

In this case, if you run "Update AL-Go System Files" it would add "AB#123456" to the end of the commit message. Additionally, you'd be able to add GitHub labels to the Pull Request and set the pull request to Auto Complete. 

Would something like this work in your scenario? Or is the commit message going to be different every time you run "Update AL-Go System Files"?

**Reply by aholstrup1** [reply]
The settings I mentioned above have been rolled out in preview now. Let us know if it's helpful to you and your use-case. Now, if you add the following to your .github/AL-Go-Settings.json file: 
```
"commitOptions": {
    "messageSuffix": "Related to XXX-1234"
}
```
It should append every commit coming from AL-Go with your message. 

If you want a different message for different workflows, you can add the setting the the workflow-specific settings files e.g. ".github\Update AL-Go System Files.settings.json". That way you can append one message to "Update AL-Go System Files" commits and another message to commits coming from "Increment Version Number" for example. 

⚠️  What you **can't** do right now is to specify the message as an input.

Does this work in your scenario? Or would you need to be able to specify the message as an input?

**Reply by TKapitan** [reply]
Thanks @aholstrup1, this is very useful and we will use it with Azure Boards!
